### PR TITLE
fix hpc combine error (close #2946)

### LINKED ISSUE #2946: combining hpc reports is failing on circleci
The error hpc combine is
```
hpc: the flag -N1 requires the program to be built with -threaded
```

### PR DESCRIPTION
**Pull request by nizar-m**
fix #2946 
### Description

Combining hpc reports fail with the following error
```
hpc: the flag -N1 requires the program to be built with -threaded
```

### Affected components 

- Build System

### Solution and Design
unset GHRTS during hpc combine, and restore after hpc combine

### Steps to test and verify

Check the hpc combine step of circleci

### CODE REVIEW

**Comment by nizar-m** [comment]
`hpc combine` is working after removing the threads configuration.

File: server/tests-py/test_graphql_queries.py [hunk 1]
```diff
@@ -263,6 +263,7 @@ def test_staff_passed_students(self, hge_ctx, transport):
     def test_user_query_auction(self, hge_ctx, transport):
         check_query_f(hge_ctx, self.dir() + '/user_query_auction.yaml', transport)
 
+    @pytest.mark.xfail(reason="Query caching")
```
Comment (jberryman) [inline server/tests-py/test_graphql_queries.py]: Could you comment with a link back to the open issue: https://github.com/hasura/graphql-engine-internal/issues/252

File: server/tests-py/test_graphql_queries.py [hunk 2]
```diff
@@ -263,7 +263,7 @@ def test_staff_passed_students(self, hge_ctx, transport):
     def test_user_query_auction(self, hge_ctx, transport):
         check_query_f(hge_ctx, self.dir() + '/user_query_auction.yaml', transport)
 
-    @pytest.mark.xfail(reason="Query caching")
+    @pytest.mark.xfail(reason="Refer https://github.com/hasura/graphql-engine-internal/issues/252")
```
Comment (jberryman) [inline server/tests-py/test_graphql_queries.py]: thanks

File: server/tests-py/test_graphql_queries.py (hunk 1 above)
Comment (nizar-m) [reply server/tests-py/test_graphql_queries.py]: Reference to the above mentioned issue has been provided.

**Pull request merged**